lib/dicom_net/association.ex

```diff
@@ -199,4 +232,64 @@ defmodule DicomNet.Association do
 
     response_ds
   end
+
+  defp cfind_response_header(command, presentation_context_id, status) do
+    asci_de = DataSet.fetch!(command, :AffectedSOPClassUID)
+    mid_de = DataSet.fetch!(command, :MessageID)
+
+    response =
+      Dicom.DataSet.from_keyword_list(
+        AffectedSOPClassUID: DataElement.value(asci_de),
+        CommandField: 0x8020,
+        MessageIDBeingRespondedTo: DataElement.value(mid_de),
+        CommandDataSetType: 0x0001, # This field shall be set to the value of 0101H (Null) if no Data Set is present; any other value indicates a Data Set is included in the Message.
+        Status: status # 0x0000 Success, 0xff00 Pending
+      )
+
+    Dicom.BinaryFormat.serialize_command_data_set(response)
+      |> Pdu.new_data_pdu(presentation_context_id) 
+      |> Pdu.serialize()
+  end
+
+  defp cfind_response_data(presentation_context_id, response) do
+    Dicom.BinaryFormat.serialize(response, [endianness: :little, explicit: true])
+      |> Pdu.new_data_pdu(presentation_context_id, 0x02)
+      |> Pdu.serialize()
+  end
+
+  defp cfind_response_tail(affectedsopclassuid, messageidbeingrespondedto, status) do
+      Dicom.DataSet.from_keyword_list(
+        AffectedSOPClassUID: DataElement.value(affectedsopclassuid),
+        CommandField: 0x8020,
+        MessageIDBeingRespondedTo: DataElement.value(messageidbeingrespondedto),
+        CommandDataSetType: 0x0101, # This field shall be set to the value of 0101H (Null) if no Data Set is present; any other value indicates a Data Set is included in the Message.
+        Status: status  # 0xff0 Success, 0xff00 Pending
+      )
+  end
+
+  defp handle_cfind(socket, command, data_set, presentation_context_id, ts_options, state) do
+    asci_de = DataSet.fetch!(command, :AffectedSOPClassUID)
+    mid_de = DataSet.fetch!(command, :MessageID)
+
+    # state.getresponses is a callback function that must be defined
+    # by the specific implementation. It must return a stream
+    # of Dicom.DataSet.from_keyword_list.
+    stream = state.getresponses.(data_set)
+    Enum.map(stream, fn response ->
+        headerbuffer = cfind_response_header(command, presentation_context_id, 0xff00)
+        :gen_tcp.send(socket, headerbuffer)
+        databuffer = cfind_response_data(presentation_context_id, response)
+        :gen_tcp.send(socket, databuffer)
+    end)
+
+    response_tail = cfind_response_tail(asci_de, mid_de, 0x0000)
```

Comment:
Now this is also not really nice, since the majority of the response is sent here, and the last part in another method. But I also need to think about how to clean this up, so OK for now.